protocol/test/core/ascend-zaps-test.ts

```diff
@@ -0,0 +1,557 @@
+import { config } from 'dotenv';
+import { ethers, waffle, network } from 'hardhat';
+import { blockTimestamp, toAtto } from '../helpers';
+import { Signer, BigNumber } from 'ethers';
+import {
+  TempleERC20Token,
+  TempleERC20Token__factory,
+  ERC20,
+  ERC20__factory,
+  Vault,
+  Vault__factory,
+  Exposure__factory,
+  JoiningFee,
+  JoiningFee__factory,
+  VaultedTemple,
+  VaultedTemple__factory,
+  AscendZaps,
+  AscendZaps__factory,
+  IBalancerVault,
+  Exposure,
+} from '../../typechain';
+import { expect } from 'chai';
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/signers';
+import { advanceTimeAndBlock } from '../utils/time';
+import { impersonateAccount } from '../utils/account';
+import { takeSnapshot, revertToSnapshot } from '../utils/snapshot';
+import constants from '../constants';
+
+config();
+
+describe.only('Ascend Zaps', async () => {
+  let TEMPLE: TempleERC20Token;
+  let DAI: ERC20;
+  let vault: Vault;
+  let joiningFee: JoiningFee;
+  let vaultedTemple: VaultedTemple;
+  let ascendZaps: AscendZaps;
+  let templeExposure: Exposure;
+
+  let owner: Signer;
+  let alan: SignerWithAddress;
+  let ben: Signer;
+  const periodDuration = 60 * 10;
+  const balanceVaultAddress = '0xBA12222222228d8Ba445958a75a0704d566BF2C8';
+
+  let snapshotId: string;
+
+  before(async function () {
+    this.timeout(0);
+    await network.provider.request({
+      method: 'hardhat_reset',
+      params: [
+        {
+          forking: {
+            jsonRpcUrl: process.env.MAINNET_RPC_URL || '',
+            blockNumber: 15674900,
+          },
+        },
+      ],
+    });
+
+    [owner, alan, ben] = await ethers.getSigners();
+
+    TEMPLE = await TempleERC20Token__factory.connect(
+      '0x470EBf5f030Ed85Fc1ed4C2d36B9DD02e77CF1b7',
+      owner
+    );
+    await impersonateAccount(constants.accounts.MULTISIG);
+    const multisig = await ethers.getSigner(constants.accounts.MULTISIG);
+    await TEMPLE.connect(multisig).addMinter(await owner.getAddress());
+
+    DAI = await ERC20__factory.connect(constants.tokens.DAI, owner);
+
+    joiningFee = await new JoiningFee__factory(owner).deploy(toAtto(0.0001));
+
+    templeExposure = await new Exposure__factory(owner).deploy(
+      'temple exposure',
+      'TPL-VAULT-EXPOSURE',
+      TEMPLE.address
+    );
+
+    vaultedTemple = await new VaultedTemple__factory(owner).deploy(
+      TEMPLE.address,
+      templeExposure.address
+    );
+
+    await templeExposure.setLiqidator(vaultedTemple.address);
+
+    vault = await new Vault__factory(owner).deploy(
+      'Temple 1m Vault',
+      'TV_1M',
+      TEMPLE.address,
+      templeExposure.address,
+      vaultedTemple.address,
+      periodDuration,
+      60,
+      { p: 1, q: 1 },
+      joiningFee.address,
+      await blockTimestamp()
+    );
+
+    await templeExposure.setMinterState(vault.address, true);
+
+    await expect(
```

Comment:
you can remove these zero checks for constructor arguments. I also suggest you remove zero address checks from solidity contract constructor.